designs/node-upgrades.md

```diff
@@ -0,0 +1,185 @@
+# Karpenter Node Upgrades - Automatic Upgrades
```

Comment:
I can see it both ways. In a way, Drift is the symptom and Upgrading is the cure. I could understand using Detection & Remediation for this, but I think Upgrading gets that across more tersely, regardless of the underlying implementation.

Reply:
The reconciliation could be replacement with an older AMI. The term "upgrading" has implications which might not be desirable.

Reply:
In what ways? We'll always use the latest AMI out of the ones specified by the user

Reply:
I would explicitly call it AMI upgrades and not specifically not call it Node Upgrades. Node upgrades suggest the K8s version is increasing, in my mind. Update may also be more suitable word than upgrade.

Reply:
The user might have changed the specification to refer to an older AMI than previously. This would typically be when the newer AMI shows unexpected behavior.

Reply:
AWS has previously called something like this an [instance refresh](https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-instance-refresh.html). Not sure if this comment is helpful, but just adding another name option 😄